tests/insertion.rs

```diff
@@ -404,6 +404,102 @@ fn after_removal() {
         .single(client_app.world());
 }
 
+#[test]
+fn dont_replicate_after_connect() {
+    let mut server_app = App::new();
+    let mut client_app = App::new();
+    for app in [&mut server_app, &mut client_app] {
+        app.add_plugins((
+            MinimalPlugins,
+            RepliconPlugins.set(ServerPlugin {
+                tick_policy: TickPolicy::EveryFrame,
+                replicate_after_connect: false,
+                ..Default::default()
+            }),
+        ))
+        .replicate::<TableComponent>()
+        .replicate::<SparseSetComponent>();
```

Comment:
This is so that we can test this later:
```rs
    client_app
        .world_mut()
        .query_filtered::<(), With<TableComponent>>()
        .single(client_app.world());
    client_app
        .world_mut()
        .query_filtered::<(), With<SparseSetComponent>>()
        .single(client_app.world());
```
The entity with `TableComponent` was spawned before replication started, then replicated to the client after the server started replication for him. The entity with `SparseSetComponent` was spawned after replication was already started.
This test is meant to ensure that both entities were replicated correctly, both before and after replication was explicitly enabled.

Reply:
Ah, I get it!
Then let's split it into two separate tests. It's better to duplicate a little code, but have different tests that test different things. Otherwise it's just not very obvious :(

Reply:
I'm happy to split it into multiple tests, but just to make it clear the reason I'm using both `TableComponent` and `SparseSetComponent` is so that I can have two different filters for two different entities. Like, if I spawned two entities both with `TableComponent`, then `query_filtered::<TableComponent>().single(..)` would find two entities, and fail. So I need something extra to differentiate the two entities by, and for that I just used `TableComponent`/`SparseSetComponent`.

Reply:
Yes, yes, I understand why, I just suggesting to this into two tests to make it even more clear what we are testing :)

Reply:
Split the test